mitodl/unified-ecommerce
# Redefine APISIX configuration in yaml

### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/5724

### Description (What does it do?)

This moves the APISIX configuration into a set of configuration files. Prior to this, APISIX had to be configured using API calls. 

This nets us a few benefits:
- Standing up the service is easier - you set some parameters in your `.env` file and go, rather than having to create a script and run it. 
- Actually makes APISIX more configurable, since we're now surfacing all three core APISIX config files in the codebase (and the deployment mode change means they get scanned for changes on a pretty aggressive schedule).
- Pulls together the APISIX configuration in a manner that means it _may_ be easier to deploy. 
- Eliminates a dependency and a container for local deployments - this doesn't use an `etcd` anymore.

This also makes some minor fixes to the routing (the unauthenticated routes needed wildcards).

### How can this be tested?

**Before testing,** `docker compose down api` first - this should trash the APISIX containers but leave the rest of your stuff intact. 

Set up your `.env` file as specified in the README. This will also serve as testing the documentation, which has changed to reflect the changes made in the PR. 

Bring up your instance. The `api` container should run successfully, and you should get some additional output by default now (as the default is to turn debugging mode on). You should be able to get to the API schema docs - e.g., http://ue.odl.local:9080/api/schema/swagger-ui/ - without having to log in. Going to the `/` route should bounce you through Keycloak successfully.
